src/base/utils/os.h

```diff
@@ -0,0 +1,65 @@
+/*
+ * Bittorrent Client using Qt and libtorrent.
+ * Copyright (C) 2018  Vladimir Golovnev <glassez@yandex.ru>
+ * Copyright (C) 2014  sledgehammer999 <sledgehammer999@qbittorrent.org>
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * In addition, as a special exception, the copyright holders give permission to
+ * link this program with the OpenSSL project's "OpenSSL" library (or with
+ * modified versions of it that use the same license as the "OpenSSL" library),
+ * and distribute the linked executables. You must obey the GNU General Public
+ * License in all respects for all of the code used other than "OpenSSL".  If you
+ * modify file(s), you may extend this exception to your version of the file(s),
+ * but you are not obligated to do so. If you do not wish to do so, delete this
+ * exception statement from your version.
+ */
+
+#pragma once
+
+#include <QtGlobal>
+
+#ifdef Q_OS_WIN
+#include <windows.h>
+#if (NTDDI_VERSION >= NTDDI_VISTA)
+#define WIN_USE_DEFAULT_PROGRAMS
+#endif
+#endif
+
+namespace Utils
+{
+    namespace OS
+    {
+        bool isTorrentFileAssocSet();
```

Comment:
Did not find Linux implementations for these two.

Reply:
Maybe you would opt to add implementations instead of removing the functions. Here is what needs to be done:
1. Query associations: launch`xdg-mime query default <mime type>`. Should return "qbittorrent.desktop" if qBittorrent is the default.
2. To set the default launch: `xdg-mime default <mime type> qbittorrent.desktop`.

The mime types are application/x-bittorrent and x-scheme-handler/magnet

Reply:
Then put them under `#ifdef`, please.

Reply:
Hell, I'll just put them back.